vagrant/box_check.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+if [ $# -eq 0 ]; then
+	exit 1
+fi
+
+boxNames=($(vagrant box list))
+
+boxNamePrefix=$(echo "$1" | sed 's/ie.*//')
+if [[ $boxNamePrefix == "" ]]; then
+	boxNamePrefix="win7"
+fi
+boxNameSuffix=$(echo "$1" | sed 's/^.*ie/ie/')
+fullBoxName="$boxNamePrefix-$boxNameSuffix"
+
+if [[ $2 == "" ]]; then
+	hasBox=1
+	for i in "${boxNames[@]}"
+	do
+		if [[ "$fullBoxName" =~ "$i" ]]; then
+			hasBox=0
+			break
+		fi
+	done
+
+	if [[ $hasBox == 1 ]]; then
+
+		boxURLs=( "http://aka.ms/vagrant-win7-ie8" "http://aka.ms/vagrant-win7-ie9" "http://aka.ms/vagrant-win7-ie10" "http://aka.ms/vagrant-win7-ie11" "http://aka.ms/vagrant-win8-ie10" "http://aka.ms/vagrant-win81-ie11" )
+		boxURL=""
+		for url in "${boxURLs[@]}"
+		do
+			if [[ $url =~ "$fullBoxName" ]]; then
```

Comment:
This checks to see if the `$fullBoxName` (e.g. `win7-ie8`) is a substring of the current URL in the array, `boxURLs`. It's like a hacked hash table of sorts since Bash 3 doesn't support associative arrays. Ideally, we'd want something like this: